source/io/module_javaScriptInvoke.js

```diff
@@ -131,156 +138,123 @@ var assignJavaScriptInvoke;
-                isValidReturnType: function (value) {
-                    return value instanceof Int8Array;
-                }
-            },
-            ArrayInt16: {
-                reader: Module.eggShell.dataReadTypedArray,
-                writer: Module.eggShell.dataWriteTypedArray,
-                isValidReturnType: function (value) {
-                    return value instanceof Int16Array;
-                }
-            },
-            ArrayInt32: {
-                reader: Module.eggShell.dataReadTypedArray,
-                writer: Module.eggShell.dataWriteTypedArray,
-                isValidReturnType: function (value) {
-                    return value instanceof Int32Array;
-                }
-            },
-            ArrayUInt8: {
-                reader: Module.eggShell.dataReadTypedArray,
-                writer: Module.eggShell.dataWriteTypedArray,
-                isValidReturnType: function (value) {
-                    return value instanceof Uint8Array;
-                }
-            },
-            ArrayUInt16: {
-                reader: Module.eggShell.dataReadTypedArray,
-                writer: Module.eggShell.dataWriteTypedArray,
-                isValidReturnType: function (value) {
-                    return value instanceof Uint16Array;
-                }
-            },
-            ArrayUInt32: {
-                reader: Module.eggShell.dataReadTypedArray,
-                writer: Module.eggShell.dataWriteTypedArray,
-                isValidReturnType: function (value) {
-                    return value instanceof Uint32Array;
-                }
-            },
-            ArraySingle: {
-                reader: Module.eggShell.dataReadTypedArray,
-                writer: Module.eggShell.dataWriteTypedArray,
-                isValidReturnType: function (value) {
-                    return value instanceof Float32Array;
-                }
-            },
-            ArrayDouble: {
-                reader: Module.eggShell.dataReadTypedArray,
-                writer: Module.eggShell.dataWriteTypedArray,
-                isValidReturnType: function (value) {
-                    return value instanceof Float64Array;
-                }
-            },
-            JavaScriptRefNum: {
-                reader: dataReadJavaScriptRefNum,
-                writer: dataWriteJavaScriptRefNum,
-                isValidReturnType: function () {
-                    return true;
+        var createJavaScriptInvokePeeker = function () {
+            var visitNumeric = function (valueRef) {
+                return Module.eggShell.readDouble(valueRef);
+            };
+
+            var visitNumeric64 = function (valueRef) {
+                return JSON.parse(Module.eggShell.readJSON(valueRef));
+            };
+
+            return {
+                visitInt8: visitNumeric,
+                visitInt16: visitNumeric,
+                visitInt32: visitNumeric,
+                visitInt64: visitNumeric64,
```

Comment:
Only include types that were supported previously (we should add more types when supported in the editor)

Reply:
ie remove the 64bits, the enums, and jsObjectrefnums

Reply:
Removed 64-bits, enums. We should talk about removing visitJSObjectRefnum from peeker and poker.

Reply:
Talked about removing visitJSObjectRefnum and will remove in a separate PR